dashboard/app.py

```diff
@@ -0,0 +1,318 @@
+import streamlit as st
+import pandas as pd
+import numpy as np
+import plotly.express as px
+import seaborn as sns
+import matplotlib.pyplot as plt
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+from datetime import datetime
+import re
+import os
+import json  
+import itables.interactive
+from itables import show
+from IPython.core.display import display, display_html, Javascript, HTML
+
+import streamlit.components.v1 as components
+
+
+from pandas import json_normalize 
+
+
+DATA_URL = ('/mnt/ssardina-pacman/cosc1125-1127-AI/www/feedback-final/stats-archive')
+DEPLOYED_URL = 'http://118.138.246.177/feedback-final'
+ORGANIZER = 'RMIT AI - COSC1125/1127'
+
+FORMAT_DATE_FILE = '.*(\d{4}-\d{2}-\d{2}-\d{2}-\d{2}).*\.json'
+
+
+def get_date_from_json(filename):
+    """
+    Extracts the date part of the json stat file
+    :param filename:
+    :return:
+    """
+    return re.match(FORMAT_DATE_FILE, filename).group(1)
+
+def get_id_from_json(filename):
+    """
+    Extracts the date part of the json stat file
+    :param filename:
+    :return:
+    """
+    return re.match('stats_(.*).json', filename).group(1)
+
+
+def normalize(x, min, max, min_new, max_new):
+    """
+    Normalize x in range (min, max) into new range (min_new, max_new)
+    :param x:
+    :param min:
+    :param max:
+    :param min_new:
+    :param max_new:
+    :return:
+    """
+    return (max_new - min_new) // (max - min) * (x - min) + min_new
+
+
+# Streamlit encourages well-structured code, like starting execution in a main() function.
+def main():
+    max_width_layout()
+
+    # Load Data
+    json_files = sorted([f for f in os.listdir(DATA_URL) if os.path.isfile(os.path.join(DATA_URL, f))], reverse=True)
+
+    df_all_games, df_all_stats = load_data( json_files )
+
+
+    # Side bar
+    json_selectbox = st.sidebar.selectbox(
+        "Select Tournament",
+        options=json_files,
+        index=0,
+        #format_func=lambda x: datetime.strptime(x, 'stats_%Y-%m-%d-%H-%M.json')
+        format_func=lambda x: get_id_from_json(x)
+    )
+
+    table_checkbox = st.sidebar.checkbox('Show Table',value=True)
+    teams_progress_checkbox = False
+    # teams_progress_checkbox = st.sidebar.checkbox('Show Teams Progress Chart',value=True)
```

Comment:
Uncomment, I had it commented out but we want it in th efull system.

Maybe add boolean constants in `config.py` to allow user to turn on/off parts of Dashborad

Reply:
Resolved and factored out to config.py to enable/disable